### DISCUSSION TITLE
v3.1.0

### DISCUSSION THREAD

**Discussion by Daveiano** [Daveiano/weewx-wdc#138] [Announcements]
- Live updates via MQTT driven stat tiles on front page GH-131
- Add slot for webcam image(s) to front page. Add optional extra webcam page GH-127 and https://github.com/Daveiano/weewx-wdc/discussions/121#discussioncomment-5280575

[MQTT Wiki](https://github.com/Daveiano/weewx-wdc/wiki/Support-for-weewx-mqtt)
[Externals Wiki](https://github.com/Daveiano/weewx-wdc/wiki/Webcams-and-Externals-Page)

## Changes made to skin.conf since 3.0.1

Line 29, following:

```diff
#radar_html = ''
#radar_heading = Recent radar

+# Set to True to have the Forecast tile and the radar/externals tile to have the same width.
+# Default is forecast = 2/3 and radar tile = 1/3.
+forecast_radar_equal_width = False

+[[mqtt]]
+     mqtt_websockets_enabled = 0
+     mqtt_websockets_host = "localhost"
+     mqtt_websockets_port = 9001
+     mqtt_websockets_ssl = 0
+     mqtt_websockets_topic = "weather/loop"

+# Include various external sources (eg. webcams) here.
+#[[external_1]]
+#    source = '<img src="http://your-server.com/uploads/webcam01.jpg" />'
+#    title = Webcam 1
+#    title_long = "Webcam 1, facing North"

+#[[external_2]]
+#    source = '<img src="http://your-server.com/uploads/webcam02.jpg" />'
+#    title = Webcam 2
+#    title_long = "Webcam 2, facing South"

+#[[external_3]]
+#    source = '<img src="http://your-server.com/uploads/webcam01.gif" />'
+#    title = Webcam 1
+#    title_long = "Webcam 1, Timelapse"

+#[[external_4]]
+#    source = '<img src="http://your-server.com/uploads/webcam02.gif" />'
+#    title = Webcam 2
+#    title_long = "Webcam 2, Timelapse"

[[forecast_zambretti]]
   enable = False
```

Line 508:

```diff
[[[about]]]
   template = about.html.tmpl
   title = About
+#[[[externals]]]
+   #template = externals.html.tmpl
+   #title = Webcams/Externals
#[[[DWD]]]
   #template = dwd.html.tmpl
   #title = Vorhersage vom DWD
```

Line 567:

```diff
[CopyGenerator]
-    copy_once = dist/main.js, dist/main.css, plotly-custom-build.min.js, favicon.ico, manifest.json, icon-192x192.png, icon-256x256.png, icon-384x384.png, icon-512x512.png, service-worker.js, offline.html, dist/assets
+    copy_once = dist/main.js, dist/main.css, plotly-custom-build.min.js, dist/live-updates.js, favicon.ico, manifest.json, icon-192x192.png, icon-256x256.png, icon-384x384.png, icon-512x512.png, service-worker.js, offline.html, dist/assets
```

## What's Changed
* Bump webpack from 5.73.0 to 5.76.0 by @dependabot in https://github.com/Daveiano/weewx-wdc/pull/132
* Add slot for webcam image/add optional extra webcam page by @Daveiano in https://github.com/Daveiano/weewx-wdc/pull/134
* MQTT driven stat tiles on index page by @Daveiano in https://github.com/Daveiano/weewx-wdc/pull/131

**Full Changelog**: https://github.com/Daveiano/weewx-wdc/compare/v3.0.1...v3.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Daveiano/weewx-wdc/releases/tag/v3.1.0'>v3.1.0</a>.</em>

**Comment by nifoc** [comment]
Thanks for implementing MQTT tiles 🥳🥳🥳

Just a heads up for everyone wanting to expose Mosquitto directly to the internet (instead of going through a reverse proxy like nginx): [Websockets (over TLS) might not work in Firefox](https://github.com/eclipse/mosquitto/issues/1211)

There are some workarounds mentioned in the linked issue, but the cleanest solution (which also allows you to just use port 443) is to just use a reverse proxy.

**Comment by JonFear** [comment]
Thanks Dave,

A bit of head scratching and it appears to be working...

wlxwx.co.uk

BW

Jon

**Reply by Daveiano** [reply]
It works kind of, but it does not look like it should. Sorry, I just found a bug.

Could you please test the following:

in includes/forecast-table.inc at line 125 please change from:

```
<script
    type="module"
    src="https://1.www.s81c.com/common/carbon/web-components/version/v1.21.0/content-switcher.min.js"
  ></script>
```

to

```
  <script
    type="module"
    src="https://1.www.s81c.com/common/carbon/web-components/version/v1.25.0/content-switcher.min.js"
  ></script>
```

Then there should be tabs visible to let you switch through radar and webcams.

BW, Dave

**Reply by JonFear** [reply]
Ah That makes more sense! Works lovely now, nice feature...

Now how do I ...? Best get on with the about page!

BW

Jon

**Reply by Daveiano** [reply]
> Ah That makes more sense! Works lovely now, nice feature...

Thank you!

> Now how do I ...? Best get on with the about page!

Sorry? Btw, I added some Code examples for basic layouts here: https://github.com/Daveiano/weewx-wdc/wiki/Code-Snippets#basic-layouts
